src/EditorFeatures/Core.Wpf/Suggestions/AsyncSuggestedActionsSource.cs

```diff
@@ -111,41 +111,123 @@ private partial class AsyncSuggestedActionsSource : SuggestedActionsSource, IAsy
+                                                pendingLowPrioritySets.Add(set);
+                                                continue;
+
+                                            default:
+                                                throw ExceptionUtilities.UnexpectedValue(set.Priority);
+                                        }
+
+                                        break;
+
+                                    case CodeActionRequestPriority.Low:
+                                        switch (set.Priority)
+                                        {
+                                            case SuggestedActionSetPriority.High:
+                                            case SuggestedActionSetPriority.Medium:
+                                            case SuggestedActionSetPriority.Low:
+                                                break;
+
+                                            default:
+                                                throw ExceptionUtilities.UnexpectedValue(set.Priority);
+                                        }
+
+                                        break;
+
+                                    case CodeActionRequestPriority.Lowest:
+                                        // We only expect suppression/configuration fixes for 'Lowest' request priority.
+                                        // These action sets have 'SuggestedActionSetPriority.None'.
+                                        Contract.ThrowIfFalse(set.Priority == SuggestedActionSetPriority.None);
+                                        break;
+
+                                    case CodeActionRequestPriority.None:
+                                        // Add all the sets for 'None' request priority.
+                                        break;
+
+                                    default:
+                                        throw ExceptionUtilities.UnexpectedValue(priority);
                                 }
+
+                                // We have an action set with matching or higher priority compared to the request priority,
+                                // so add it to the collector set.
+                                currentActionCount += set.Actions.Count();
+                                collector.Add(set);
                             }
 
-                            if (priority == CodeActionRequestPriority.Normal)
+                            // Drain any pending actions sets for the next request priority which were computed for a higher request priority.
+                            var pendingSetsToDrain = priority switch
+                            {
+                                CodeActionRequestPriority.High => pendingMediumPrioritySets,
+                                CodeActionRequestPriority.Normal => pendingLowPrioritySets,
+                                _ => null,
+                            };
```

Comment:
this seems odd, esp. now that you're supporting all the pri classes.  My expectation is that we would just drain these when we're operating at the priority level the pending set corresponds to.  so once we hit normalPriority, we drain the pendingNormalPriSet, once we hit low pri, we drain the pendingLowPriSet.

Note: `CodeActionRequestPriority.High => pendingMediumPrioritySets,` doesn't even seem to make sense to me.  If we're in the high pri bucket, why would we add normal pri items to it?